pkg/storage/s3/s3.go

```diff
@@ -66,11 +70,18 @@ func NewDriver(crname string, crnamespace string, c *opapi.ImageRegistryConfigSt
 	}
 }
 
-// GetType returns the name of the storage driver
+// UpdateFromStorage updates the driver from an image registry resource
+func (d *driver) UpdateFromStorage(cfg opapi.ImageRegistryConfigStorage) {
+	d.Config = cfg.S3
```

Comment:
and you should probably consider making this a deepcopy too, lest you accidentally assign it elsewhere.

Reply:
updated